### PR TITLE
Improvement of toggle buttons borders

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Related issues

Closes second part of #2163 

### Description

Coming from the work on font prototype where we now have multiple single toggle buttons not included in a `.btn-group`:

```html
  <input type="checkbox" class="btn-check" id="btn-check" autocomplete="off">
  <label class="btn btn-primary mb-5" for="btn-check" onclick="toggleLines()">Toggle yellow lines display</label>

  <input type="checkbox" class="btn-check" id="btn-bg" autocomplete="off">
  <label class="btn btn-primary mb-5" for="btn-bg" onclick="toggleBackground()">Toggle red background display</label>
```

Simple solution was to make the selector a bit more specific.

### Motivation & Context

Before the change:
screenshot-localhost_9001-2023 09 13-15_40_19

After the change:
screenshot-localhost_9001-2023 09 13-15_39_35

### Types of change

- Bug fix (non-breaking which fixes an issue)

### Live previews

- <https://deploy-preview-2239--boosted.netlify.app/docs/5.3/forms/checks-radios/#toggle-buttons/>

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

#### Accessibility

- (na) My change follows accessibility good practices; I have at least run axe

#### Design

- (na) My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- (na) My change is compatible with a responsive display

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
- (na) I have added JavaScript unit tests to cover my changes
- (na) I have added SCSS unit tests to cover my changes

#### Documentation

- (na) My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- (na) My change introduces changes to the migration guide
- (na) My new component is well displayed in [Storybook](https://deploy-preview-2239--boosted.netlify.app/storybook)
- [x] My new component is compatible with RTL
- [x] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
- [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
- [x] Code review
- (na) Design review
- (na) A11y review

#### After the merge

- [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by MewenLeHo** [comment]
Not sure if an update of the migration guide is needed or not 🤔

File: scss/forms/_form-check.scss [hunk 1]
```diff
@@ -262,14 +262,6 @@
       height: 100%;
       content: "";
     }
-
-    &:not(:first-of-type) {
-      border-left: 0;
-    }
-
-    &:not(:last-of-type) {
```
Comment (louismaximepiton) [inline scss/forms/_form-check.scss:270]: Maybe we can keep those blocks but update the selector to `.btn-group &:not(...)` ?
Seems to work on my side and would more understandable to me.

**Review by louismaximepiton — APPROVED** [review]
I think there's no need to have a migration note on this one.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
There's a regression in the second example of https://deploy-preview-2239--boosted.netlify.app/docs/5.3/forms/checks-radios/#with-icon with the hover style:

**Comment by louismaximepiton** [comment]
Ah yes Good catch my bad on this.

**Review by louismaximepiton — APPROVED** [review]
Couldn't find any other breaking change. Can't find of another way to achieve this behavior.

**Review by julien-deramond — APPROVED** [review]

**Pull request merged**